subsys/fs/fs.c

```diff
@@ -336,13 +407,8 @@ int fs_opendir(struct fs_dir_t *zdp, const char *abs_path)
 
 	if (strcmp(abs_path, "/") == 0) {
 		/* Open VFS root dir, marked by zdp->mp == NULL */
-		k_mutex_lock(&mutex, K_FOREVER);
```

Comment:
This is not correct mutext to use. This one is used for protecting mount points so that when path to FS is figured out the mount list do not get removed/added.
Using this mutex for ops like fs_open means that you are globally locking VFS to perform operation of some mount point, so if you take a lot of time to write to one mount point the other thread has to wait on own fs_ op, even if that is applied to different mount point.